rep-0144.rst

```diff
@@ -0,0 +1,139 @@
+  overly verbose
+* using catchall names such as ``utils`` should be avoided as they do not scope what goes
+  into the package or what should be outside the package
+* a package name should not contain ``ros`` as it is redundant.
+  Exceptions include core packages and ROS bindings of an upstream library
+  (e.g. ``moveit_ros``)
+* one of ROS's goals is to develop a canonical set of tools for making robots do
+  interesting things.
+  The package name should describe what the package does, not where it came from.
+  Then again, as stated in the rules below, ``if a package is specialized
+  by an entity (lab, company, ...), prepend the name of the entity``.
+  But once the package is commonly used, owned and maintained, that name can be dropped
+  as the package becomes the reference
+* to check whether a name is taken, consult [3]_. If you'd like your
+  repository included in that list, see the tutorial at [4]_
+
+Naming Rules
+------------
+
+The following rules define the different parts of the package name.
+The overall idea is to prepend a name with words that distinguish it from similar 
+functional implementation (e.g.: ``lab``, ``robot``) but in order of importance 
+(e.g.: ``python_robot_lab``).
+Similarly, words that specialize this functionality are appended
+(e.g. ``msgs``, ``config`` ...).
+
+The rules to add those words should be followed in order.
+For prefixes:
+
+* if a package is specialized by an entity (lab, company, ...), prepend the 
+  name of the entity.
+  Once the package is commonly used, owned and maintained, that name can be dropped
+* if a package is specialized for a software project, prepend its name
+* if a package is specialized for a hardware piece, prepend its name
+* if a package is specialized for a robot, prepend its name
+
+For suffixes:
+
+* if a package is a driver, append ``driver``
+* if a package contains any of a ROS message/service/action, append ``msgs``
+* if a package is a plugin for a library, append ``<library_name>_plugins``, e.g. 
+  ``pr2_gazebo_plugins``
+
+Special Suffixes:
+
+* a meta package for a robot should be named ``<name_of_the_robot>_robot``, e.g. 
+  ``pr2_robot``
+* a package containing the URDF and meshes of a robot should be named 
+  ``<name_of_the_robot>_description``, e.g ``pr2_description``
+* if a package is meant for test only, prepend ``test``
+
+Special Cases
+-------------
+
+* a package containing only a set of launch files should end with ``launch``
+* a package containing only a set of launch files whose goal is to start a robot
+  should end with ``bringup``
+* a package containing one or more tutorials only should end with ``tutorials``.
+  If it is a set of tutorials for another package, it should contain that other
+  package name: e.g. ``navigation`` and ``navigation_tutorials``
+* a package containing one or more demos only should end with ``demos``
+* a package containing one or more tests only should end with ``tests``
```

Comment:
About 10 lines earlier `if a package is meant for test only, prepend ``test`, which confuses with this line (`test` or `tests`).